src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -188,10 +188,12 @@ public async Task Invoke(HttpContext context)
             IDictionary<string, string> commands = this.requestParser.ParseRequestCommands(context);
             if (commands.Count > 0)
             {
-                // Strip out any unknown commands.
+                // Strip out any unknown commands
+                IEqualityComparer<string> comparer = (commands as Dictionary<string, string>)?.Comparer ?? StringComparer.OrdinalIgnoreCase;
+
                 foreach (string command in new List<string>(commands.Keys))
                 {
-                    if (!this.knownCommands.Contains(command))
+                    if (!this.knownCommands.Contains(command, comparer))
```

Comment:
I'm also fine with removing the above line and just using:
```suggestion
                    if (!this.knownCommands.Contains(command, StringComparer.OrdinalIgnoreCase))
```